### PR TITLE
[postgres] Make `convertToStringForQuery` more strict

### PR DESCRIPTION
**Pull request by nathan-artie**

### CODE REVIEW

File: lib/postgres/scanner.go [hunk 1]
```diff
@@ -176,36 +175,8 @@ func convertToStringForQuery(value any, dataType schema.DataType) (string, error
 			return "", fmt.Errorf("expected string got %T with value %v", value, value)
 		}
 		return QuoteLiteral(stringValue), nil
-	case int, int8, int16, int32, int64:
-		switch dataType {
-		case schema.Int16, schema.Int32, schema.Int64:
-			return fmt.Sprint(value), nil
-		default:
-			slog.Error("int8/16/32/64 value with non-int column type",
```
Comment (nathan-artie) [inline lib/postgres/scanner.go:184]: Haven't seen any of these errors in the logs so I think it's pretty safe to remove the `DataType` check.

File: lib/postgres/scanner.go [hunk 2]
```diff
@@ -176,36 +175,8 @@ func convertToStringForQuery(value any, dataType schema.DataType) (string, error
...
-				slog.Any("value", value),
-				slog.Any("dataType", dataType),
-			)
-		}
-	case float32, float64:
-		switch dataType {
-		case schema.Real, schema.Double:
-			return fmt.Sprint(value), nil
-		default:
-			slog.Error("float32/64 value with non-float column type",
```
Comment (nathan-artie) [inline lib/postgres/scanner.go:194]: Ditto https://github.com/artie-labs/reader/pull/278/files#r1520679713

File: lib/postgres/scanner.go [hunk 3]
```diff
@@ -176,36 +175,8 @@ func convertToStringForQuery(value any, dataType schema.DataType) (string, error
...
-				slog.Any("value", value),
-				slog.Any("dataType", dataType),
-			)
-		}
-	case bool:
-		switch dataType {
-		case schema.Bit, schema.Boolean:
-			return fmt.Sprint(value), nil
-		default:
-			slog.Error("bool value with non-bool column type",
```
Comment (nathan-artie) [inline lib/postgres/scanner.go:204]: Ditto https://github.com/artie-labs/reader/pull/278/files#r1520679713

File: lib/postgres/scanner_test.go [hunk 4]
```diff
@@ -148,14 +148,14 @@ func TestConvertToStringForQuery(t *testing.T) {
 
 func TestScanTableQuery(t *testing.T) {
 	primaryKeys := []primary_key.Key{
-		{Name: "a", StartingValue: "1", EndingValue: "4"},
-		{Name: "b", StartingValue: "2", EndingValue: "5"},
+		{Name: "a", StartingValue: int64(1), EndingValue: int64(4)},
```
Comment (nathan-artie) [inline lib/postgres/scanner_test.go:151]: These should have been `int64` all along.

File: lib/postgres/scanner_test.go [hunk 5]
```diff
@@ -148,14 +148,14 @@ func TestConvertToStringForQuery(t *testing.T) {
...
+		{Name: "b", StartingValue: int64(2), EndingValue: int64(5)},
 		{Name: "c", StartingValue: "3", EndingValue: "6"},
 	}
 	cols := []schema.Column{
 		{Name: "a", Type: schema.Int64},
 		{Name: "b", Type: schema.Int64},
-		{Name: "c", Type: schema.Int64},
+		{Name: "c", Type: schema.Text},
```
Comment (nathan-artie) [inline lib/postgres/scanner_test.go:158]: Changed this so we have a test for a non-int type.

**Review by Tang8330 — APPROVED** [review]

**Pull request merged**